opal/mca/accelerator/cuda/accelerator_cuda.c

```diff
@@ -435,34 +481,23 @@ static int accelerator_cuda_memcpy(int dest_dev_id, int src_dev_id, void *dest,
         return OPAL_SUCCESS;
     }
 
-    /* Async copy then synchronize is the default behavior as some applications
-     * cannot utilize synchronous copies. In addition, host memory does not need
-     * to be page-locked if an Async memory copy is done (It just makes it synchronous
-     * which is what we want anyway):
-     * https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#concurrent-execution-host-device
-     * Additionally, cuMemcpy is not necessarily always synchronous. See:
-     * https://docs.nvidia.com/cuda/cuda-driver-api/api-sync-behavior.html
-     * TODO: Add optimizations for type field */
-    result = cuMemcpyAsync((CUdeviceptr) dest, (CUdeviceptr) src, size, opal_accelerator_cuda_memcpy_stream);
-    if (OPAL_UNLIKELY(CUDA_SUCCESS != result)) {
-        opal_show_help("help-accelerator-cuda.txt", "cuMemcpyAsync failed", true, dest, src,
-                       size, result);
-        return OPAL_ERROR;
-    }
-    result = cuStreamSynchronize(opal_accelerator_cuda_memcpy_stream);
+    result = cuMemcpy((CUdeviceptr) dest, (CUdeviceptr) src, size);
```

Comment:
That's the important change I guess. The rationale was that I saw that `cuMemcpyAsync` has to pin the memory, which is expensive. Since this is not an asynchronous function I don't see the need to pay this cost. On the other hand, `cuMemcpy` synchronizes with stream 0 (AFAIU) so that's a different cost to pay...

Reply:
>  cuMemcpyAsync has to pin the memory

This would be the case if host memory isn't pinned (either using cuMemHostRegister or by allocating through cuMemAllocHost). If you're reusing host buffers, then it would probably better to pin them once and release pinnings during cleanup phase.

> On the other hand, cuMemcpy synchronizes with stream 0 (AFAIU) so that's a different cost to pay...

I would recommend against synchronizing with stream 0.

Reply:
> This would be the case if host memory isn't pinned (either using cuMemHostRegister or by allocating through cuMemAllocHost). If you're reusing host buffers, then it would probably better to pin them once and release pinnings during cleanup phase.

I do use temporary buffers in the collectives. I guess I can register them, since they are reused multiple times. For now, I will revert this change. I guess the same would apply to the hip component